File: DesignPatternsExercises/java/TemplateMethodDesignPattern/Exercise/BusTravel.java
public class BusTravel extends Travel {
    @Override
    public void setDates() {
        System.out.println("Setting dates...");
        this.startDate = "09/05/2019";
        this.endDate = "10/01/2019";
    }

    @Override
    public void setDestination() {
        System.out.println("Setting destination...");
        this.destination = "New York, NY, USA";
    }

    @Override
    void setMeal() {
    }

    @Override
    void usePrivateVehicle() {
    }

    @Override
    void bookTicket() {
        System.out.println("Booking ticket...");
    }

    @Override
    void displayTravelDetails() {
        System.out.print("Bus Travel:");
        if (booked)
            __________(2)__________.displayTravelDetails();
        else
            System.out.println(" Transaction cancelled");
        System.out.println();
    }
}

File: DesignPatternsExercises/ruby/template_method_design_pattern/solution/bus_travel.rb
require_relative 'travel'

# class BusTravel <  __________(3)__________
class BusTravel < Travel
  def dates
    puts 'Setting dates...'
    @start_date = '09/05/2019'
    @end_date = '10/01/2019'
  end

  def destination
    puts 'Setting destination...'
    @destination = 'New York, NY, USA'
  end

  def meal
  end

  def use_private_vehicle
  end

  def book_ticket
    puts 'Booking ticket...'
  end

  def display_travel_details
    print 'Bus Travel:'
    # @booked ? __________(4)__________ : (puts ' Transaction cancelled')
    @booked ? super : (puts ' Transaction cancelled')
    puts
  end
end
